build-and-push/action.yaml

```diff
@@ -0,0 +1,32 @@
+name: "build and push images"
+description: "build and push images do docker using docker compose"
+inputs:
+  location:
+    required: false
+    type: string
+  subdir:
+    required: false
+    type: string
+  tags:
+    required: false
+    type: string
+  type:
+    required: false
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: Build and push ${{ inputs.type }}
+      env:
+        BUILD_LOCATION: ${{ inputs.location }}
+        BUILD_TYPE: ${{ inputs.type }}
+        TAGS: ${{ inputs.tags }}
+      shell: bash
+      run: >-
+        cd ${{ inputs.subdir }};
+        [ -f "db/schema.rb" ] && chmod 777 db/schema.rb;
+        [ -f "bin/checkout_all.sh" ] && ./bin/checkout_all.sh;
+        docker compose build ${BUILD_TYPE} --build-arg BUILDKIT_INLINE_CACHE=1 &&
+        docker compose push ${BUILD_TYPE} &&
+        ruby -e "ENV.fetch('TAGS', []).split(',').each {|tag| %x{docker tag #{ENV['REGISTRY']}/#{ENV['REPO_LOWER']}#{ENV['BUILD_LOCATION']}:#{ENV['TAG']} #{ENV['REGISTRY']}/#{ENV['REPO_LOWER']}#{ENV['BUILD_LOCATION']}:#{tag}}; %x{docker push #{ENV['REGISTRY']}/#{ENV['REPO_LOWER']}#{ENV['BUILD_LOCATION']}:#{tag}}}"
```

Comment:
Can we make it a little more easier to read?

ruby -e "
  ENV.fetch('TAGS', []).split(',').each do |tag|
    %x{docker tag #{ENV['REGISTRY']}/#{ENV['REPO_LOWER']}#{ENV['BUILD_LOCATION']}:#{ENV['TAG']} #{ENV['REGISTRY']}/#{ENV['REPO_LOWER']}#{ENV['BUILD_LOCATION']}:#{tag}}
    %x{docker push #{ENV['REGISTRY']}/#{ENV['REPO_LOWER']}#{ENV['BUILD_LOCATION']}:#{tag}}
  end
"